mac

```diff
@@ -61,24 +65,27 @@ safe_append_to_zshrc() {
 update_shell() {
   local shell_path;
   shell_path="$(which zsh)"
-
-  fancy_echo "Changing your shell to zsh ..."
-  if ! grep "$shell_path" /etc/shells > /dev/null 2>&1 ; then
-    fancy_echo "Adding '$shell_path' to /etc/shells"
-    sudo sh -c "echo $shell_path >> /etc/shells"
+  if [ $shell_path != $SHELL ]; then
```

Comment:
This function was always run no matter even if your login shell was already `zsh`. This section of code will be skipped if it is already setup